.github/workflows/test_package.yml

```diff
@@ -0,0 +1,91 @@
+# This workflow will test and build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Test and Package
+
+on:
+  push:
+    branches: [ master ]
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    branches: [ master ]
+    paths-ignore:
+      - '**.md'
+  release:
+    types: [ prereleased, released ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare version name
+        id: prep
+        run: |
+          VERSION_TAG=edge
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION_TAG=${GITHUB_REF#refs/tags/}
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
+            VERSION_TAG=-$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+          elif [[ $GITHUB_REF == refs/pull/* ]]; then
+            VERSION_TAG=-pr-${{ github.event.number }}
+          fi
+          if [ ${VERSION_TAG} != ${GITHUB_REF#refs/tags/} ]; then
+            VERSION_TAG=v${VERSION_TAG}-sha-${GITHUB_SHA::8}-SNAPSHOT
+          fi
+          echo ::set-output name=versionTag::${VERSION_TAG}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          cache: maven
+      - name: Run Java tests
+        run: mvn test -Drevision=${{ steps.prep.outputs.versionTag }}
+      - name: Persist Java tests reports
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: Test reports
+          path: |
+            gtfs-realtime-validator-lib/target/surefire-reports
+            gtfs-realtime-validator-webapp/target/surefire-reports
+  package:
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare version name
+        id: prep
+        run: |
+          VERSION_TAG=edge
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION_TAG=${GITHUB_REF#refs/tags/}
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
+            VERSION_TAG=-$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+          elif [[ $GITHUB_REF == refs/pull/* ]]; then
+            VERSION_TAG=-pr-${{ github.event.number }}
+          fi
+          if [ ${VERSION_TAG} != ${GITHUB_REF#refs/tags/} ]; then
+            VERSION_TAG=v${VERSION_TAG}-sha-${GITHUB_SHA::8}-SNAPSHOT
+          fi
+          echo ::set-output name=versionTag::${VERSION_TAG}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          cache: maven
+      - name: Package lib/cli and web app jars with Maven
+        run: mvn -B package --file pom.xml -DskipTests -Drevision=${{ steps.prep.outputs.versionTag }}
```

Comment:
To persist artifacts to the GitHub Maven package registry, we want `mvn ... deploy` here, and then we don't need to manually upload the artifacts later in the script. 

See OTP config for an example:
https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/.github/workflows/cibuild.yml#L53

...and the GitHub documentation here:
https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-github-packages

`mvn deploy` actually deploys the artifacts to GitHub's Maven repo with other metadata - right now they are just being uploaded as raw files. I believe this will give us a more organized package system that's easier to browse (see my other comment). 

This also more closely matches the manual release workflow we were previously using:
https://github.com/CUTR-at-USF/gtfs-realtime-validator#cutr-release-process

...when manually deploying to our [CUTR Maven repo](https://github.com/CUTR-at-USF/cutr-mvn-repo). You can see the metadata that gets deployed along with the JARs by looking at this commit - https://github.com/CUTR-at-USF/cutr-mvn-repo/commit/bd1a02d2b6968cecd3ceb3fbc2bdf8cea44de65f.

Related - if we change the release process we should update that section of the README in this project.